pkg/project/project.go

```diff
@@ -219,29 +235,41 @@ func generateProjectName() string {
 	return strings.Join(str, " ")
 }
 
-// FindFileOrDirectory searches for a file or directory with name `filename`.
-// Search starts in `startDir` and will traverse through all parent directories until the file is found,
-// root directory is reached or `maxRecursiveIteration` is exceeded.
-func FindFileOrDirectory(startDir, fileDir, filename string) (string, bool) {
+// FindFileOrDirectory searches for a file or directory named `filename`.
+// Search starts in `directory` and will traverse through all parent directories.
+// `directory` may also be a file, and in that case will start from the file's directory.
+func FindFileOrDirectory(directory, filename string) (string, bool) {
 	i := 0
 	for i < maxRecursiveIteration {
-		if fileExists(filepath.Join(startDir, fileDir, filename)) {
-			return filepath.Join(startDir, fileDir, filename), true
+		if isRootPath(directory) {
+			return "", false
 		}
 
-		startDir = filepath.Clean(filepath.Join(startDir, ".."))
-		if startDir == "." || startDir == "/" || driveLetterRegex.MatchString(startDir) {
-			return "", false
+		// remove any trailing file from directory
+		directory = filepath.Dir(directory + string(filepath.Separator))
+
+		if fileExists(filepath.Join(directory, filename)) {
+			return filepath.Join(directory, filename), true
 		}
 
+		directory = filepath.Clean(filepath.Join(directory, ".."))
+
 		i++
 	}
 
-	log.Warnf("didn't find %s after %d iterations", filename, maxRecursiveIteration)
+	log.Warnf("Max %d iterations reached without finding %s", maxRecursiveIteration, filename)
 
 	return "", false
 }
 
+func isRootPath(directory string) bool {
```

Comment:
Don't think `isRootPath` is needed because it's only used once.

Reply:
It's for readability... it makes the code using `isRootPath` more readable.

Reply:
what about decorating with a comment?

Reply:
Comments are bad practice... the code should be the comment.